extensions/amp-pan-zoom/0.1/test/test-amp-pan-zoom.js

```diff
@@ -0,0 +1,265 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import '../amp-pan-zoom';
+import {htmlFor} from '../../../../src/static-template';
+
+describes.realWin('amp-pan-zoom', {
+  amp: {
+    extensions: ['amp-pan-zoom'],
+  },
+}, env => {
+  let win, doc, sandbox;
+  let el, impl;
+  let svg;
+
+  const measureMutateElementStub = (measure, mutate) => {
+    return Promise.resolve().then(measure).then(mutate);
+  };
+
+  /**
+   * This function takes an object of attributes and constructs and builds
+   * an amp-pan-zoom element and attaches it to the page. Assumes the contents
+   * to be an SVG of width and height 300x300.
+   * @param {Object} opt_attributes
+   */
+  function getPanZoom(opt_attributes) {
+    el = htmlFor(doc)`
+      <amp-pan-zoom layout="fixed" width ="300" height="400">
+      </amp-pan-zoom>
+    `;
+
+    for (const key in opt_attributes) {
+      el.setAttribute(key, opt_attributes[key]);
+    }
+
+    svg = htmlFor(doc)`
+      <svg width="100" height="100">
+        <rect width="100" height="100" fill="#95B3D7"></rect>
+      </svg>
+    `;
+
+    el.appendChild(svg);
+    doc.body.appendChild(el);
+    return el.build().then(() => {
+      impl = el.implementation_;
+      sandbox.stub(impl, 'measureMutateElement')
+          .callsFake(measureMutateElementStub);
+      sandbox.stub(impl, 'mutateElement')
+          .callsFake(mutate => measureMutateElementStub(undefined, mutate));
+    });
+  }
+
+  beforeEach(() => {
+    win = env.win;
+    doc = win.document;
+    sandbox = env.sandbox;
+    env.iframe.height = 500;
+    env.iframe.width = 400;
+  });
+
+  it('should initialize all attributes correctly', () => {
+    return getPanZoom({
+      'max-scale': '4',
+      'initial-scale': '2',
+      'initial-x': '1',
+      'initial-y': '5',
+      'reset-on-resize': '',
+      'disable-double-tap': '',
+    }).then(() => {
+      expect(impl.initialScale_).to.equal(2);
```

Comment:
Would ideally not look at the private variables in a test and instead test things that are part of the public interface (e.g. the actual scale of the element in this case). Would probably be a lot of work to change everything here, but something to keep in mind for the future.

Reply:
This is intentional. There is a test for this that tests the public interface. It's a trivial test that basically verifies that everything was initialized correctly. I guess the only thing it's testing is `getAttribute`, so we could remove it.